src/network.c

```diff
@@ -104,15 +104,13 @@ static gboolean transfer(RaucTransfer *xfer, GError **error)
 	r = curl_easy_perform(curl);
 	if (r == CURLE_HTTP_RETURNED_ERROR) {
 		g_set_error(error, G_IO_ERROR, G_IO_ERROR_FAILED, "HTTP returned >=400");
-		res = FALSE;
```

Comment:
While checking if we could replace each goto's with a `curl_easy_cleanup(curl)` + direct return, I stumbled across line 82 where we have a `goto out` without setting `error` before. We should fix this, too, either here or in a follow-up PR.

Reply:
Done.